lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2621,37 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Callback to subscribe to [ViewFocusChange] notifications.
+typedef ViewFocusChangeCallback = void Function(ViewFocusChange);
+
+/// An event to communicate view focus changes between the framework and engine.
+///
+/// If the event comes from the framework it means there was a flutter focus change.
+/// If the event comes from the engne it means there was a platform focus change.
+final class ViewFocusChange {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusChange({
+    required this.viewId,
+    required this.viewFocusState,
+  });
+
+  /// The id of the [FlutterView] that experienced the focus change.
+  final int viewId;
+
+  /// The current [FocusState] of the view.
+  final ViewFocusState viewFocusState;
```

Comment:
Even though this class has the word "change" in the name, nothing in its properties communicate a change. Can we call it just `ViewFocus`? Or for consistency with `FlutterView`, `FlutterViewFocus`?

Reply:
> Even though this class has the word "change" in the name, nothing in its properties communicate a change.
Actually it does, both the engine and the framework will keep track of different focus systems. As soon as they detect there is a change they'll use this class to communicate the deltas. This class is not intended to be used as a mechanism to query the focus state of a given flutter view. This class is meant to be used to communicate focus deltas.

1. Nothing is focused
2. View 1 gained focus (fire the event)
3. View 1 lost focus (fire the event)
4. View 2 gained focus (fire the event)
5. View 3 gained focus (fire the event)

Reply:
I guess what I'm trying to say is that this class does not actually contain the delta. Two common ways to communicate a delta is either using an `oldValue`/`newValue` pair or a `newValue`/`valueDifference` pair. However, this class only contains the `newValue` (`viewFocusState`). Therefore, it's not a "change". It's just a particular focus state at some point in time.

Sometimes the delta is implied when only two values are possible, such as `bool` or a two-value enum. Then a change notification only needs to communicate the new value. The old value is implied to be "the other one". `ViewFocusState` technically has 4 possible values, so the old value cannot be implied with full accuracy. Although, if we think that the difference between all the "focused" values is minor, then it can be thought of as a two-value enum.

Reply:
> Therefore, it's not a "change". It's just a particular focus state at some point in time.

Independently of its params, it is meant to be used to communicate differences in focus state hence it is a delta. 

> oldValue/newValue pair or a newValue/valueDifference pair.

We can easily extend the class to accept those things. I thought the intention was to make this a class so that we can easily extend it without having to make big changes in the framework/engine.

Reply:
> "Wanna get the latest state? Apply this change"

The class seems to represent the latest state, though. It is not a delta that needs to get applied to an old value or a history. It stands on its own. It _is_ the latest state (or a request to get into that state, see the other discussions).

Reply:
Thanks for the detailed explanations. I made the suggested changes. PTAL